### DISCUSSION TITLE
[Consistency Protocol] EC support

### DISCUSSION THREAD

**Discussion by ilixiaocui** [opencurve/curve#1950] [General]
Currently curve has file storage and block storage, but no object storage. Object storage generally needs to support EC, so the community started a discussion here, how curve supports the EC protocol.

An initial idea from fast-2020: Can CRaft fall into engineering practice?

**Comment by hzh0425** [comment]
We hope to design an independent EC replication system that can coexist with the original Raft system

**Comment by Tangruilin** [comment]
I agree with @hzh0425 
For EC, different CS will storage different data, which is different from raft. If we need to implement ECraft, the RaftLog may required a lot change

**Comment by WyattJia** [comment]
Looking forward to further progress.
